node/src/actors/chain_manager/mining.rs

```diff
@@ -625,6 +639,97 @@ fn build_block(
     (block_header, txns)
 }
 
+fn internal_calculate_mining_probability(
+    rf: f64,
+    n: f64,
+    first_time: bool,
```

Comment:
If you pass a boolean argument to the function, it's like having two different functions. If the code can be simplified to calculate this argument inside then great, otherwise just split it into two functions to improve clarity.

Reply:
The idea is differing the main prob and the added prob when using bf